ui/screens.py

```diff
@@ -1430,14 +1444,17 @@ def __init__(self, app, accountScreen):
         self.isPoppable = True
         self.canGoHome = True
 
+        # Currently shown agenda dropdowns are saved here.
+        self.dropdowns = []
         self.pages = []
         self.page = 0
         self.ignoreVoteIndexChange = False
+        self.voteSet = False
 
-        self.app.registerSignal(ui.POOL_SIGNAL, self.setVote)
+        self.app.registerSignal(ui.PURCHASEINFO_SIGNAL, self.setVote)
 
         self.accountScreen = accountScreen
-        self.app.registerSignal(ui.BLOCKCHAIN_CONNECTED, self.setAgendas)
+        self.app.makeThread(self.getAgendaInfosFunc(), self.setAgendas)
```

Comment:
You had made the comment at this point in the code:
> As mentioned above, let's just pull the agenda info in a makeThread here instead of relying on DcrdataBlockchain

And I misinterpreted it.

Reply:
Actually, at a different point in the code, but this is where we are pulling the data.

Reply:
Sorry for the confusion. Looks good now.

Reply:
Sorry for the confusion. It looks good now